src/Elastic/Connection/Client.php

```diff
@@ -836,4 +843,19 @@ public function releaseLock( $type ) {
 		$this->lockManager->releaseLock( $type );
 	}
 
+	/**
+	 * @since 4.2
+	 *
+	 * @return bool
+	 */
+	public function isOpenSearch(): bool {
+
+		if ( !isset( $this->distribution ) ) {
+			$info = $this->info();
+			$this->distribution = $info['version']['distribution'] ?? 'elasticsearch';
```

Comment:
Why are you storing this as a property?

Reply:
If `isOpenSearch()` is called multiple times in a single request (I don't think this is currently the case, but it seems probable that it might be in the future), we do not want to repeat the expensive `info` query.

Reply:
Right. Didn't realize a query was being executed. Makes sense.